packages/varlock-website/src/content/docs/guides/plugins.mdx

```diff
@@ -0,0 +1,105 @@
+# @plugin(@varlock/b-plugin@1.2.3)  # pinned to v1.2.3
+# @plugin(@varlock/c-plugin@^2.3.4) # use latest v2.3.x
+```
+
+:::caution[Only `@varlock/*` plugins supported for now]
+For now, only official Varlock plugins under the `@varlock` npm scope are supported. We plan to support third-party plugins in the future, along with additional plugin source types (e.g., jsr, git, http, etc.) - but need additional security measures in place first.
+:::
+
+{/*
+INFO TO ADD LATER
+- local single file
+- local path (monorepo)
+- https url
+- git
+- generic npm/jsr
+*/}
+
+## Plugin scope
+Plugins are loaded globally, and the additional functionality they provide will be available in all `.env` files in your project. Only a single `@plugin()` decorator is needed to load the plugin, even if multiple files use its functionality. If a plugin is loaded in multiple files, no error will be thrown, as long as they all use the same version.
+
+Note that plugins will not be loaded from an inactive file - for example an environment-specific file that does not match the current environment, or one that uses the [`@disable` root decorator](/reference/root-decorators/#disable).
+
+No specific namespacing or prefixes are enforced, and any naming conflicts will trigger an error, but plugins will use fairly specific names to avoid conflicts.
+
+## Initialization
+Plugins are initialized using custom root decorators that they introduce. In some cases no specific initialization is needed, and in others, you may need to intialize multiple instances of a plugin with different options, referred to by some identifier. How (or if) a plugin needs to be initialized depends on the specific plugin and can depend on the the external service's data/auth model.
+
+A plugin initialization root decorator is used to set IDs, toggle features, and wire up auth. Note that sensitive data should be passed in via references to config items within your schema.
+
+```env-spec title="Plugin initialization example" "@initOpVault"
+# @initOpVault(account=acmeco, token=$OP_TOKEN, allowAppAuth=forEnv(dev))
+# ---
+# @type=opServiceAccountToken @sensitive
+OP_TOKEN=
+```
+
+
+### Multiple plugin instances
+In secret storage tools, you should segment your data to follow the [_principle of least privilege_](https://en.wikipedia.org/wiki/Principle_of_least_privilege), so that different environments/services/devs only have access to the minimal secrets they need. At the very least, this usually means splitting your extra sensitive prod secrets from everything else, but it can be as fine-grained as needed.
+
+We cannot always assume that you won't need access to multiple segments at the same time. In these cases, a plugin may be designed to be initialized multiple times with some kind of id parameter. Resolver functions and decorators can then accept an additional parameter to specify which instance to use. It is up to the plugin author to structure this in a way that makes sense.
```

Comment:
last sentence is probably unnecessary as we're the authors for the foreseeable future